### DISCUSSION TITLE
Add support for http protocol within provider options

### DISCUSSION THREAD

**Discussion by matchatype** [nextauthjs/next-auth#2006] [Help]
## Summary 💭

Some OAuth solutions, like FusionAuth, can also be installed locally. Supporting `http` would be really helpful in the context of a fully local development environment.

- **purpose of the feature**: Supporting fully local development and testing environments.
- **potential problems**: None.
- **potential alternatives**: Generating SSL certificates for localhost.

**Comment by balazsorban44** [comment]
what keeps you back from this? could you give an example?

**Comment by matchatype** [comment]
The actual implementation of NextAuth's FusionAuth provider has `https` protocol hard-coded in all urls: `authorizationUrl`, `accessTokenUrl` and `profileUrl`. Currently you cannot easily reach something like `http://localhost:9011`, because NextAuth will point to `https://localhost:9011`.

**Comment by balazsorban44** [comment]
Have you seen the docs? https://next-auth.js.org/providers/fusionauth
You can override any of the options you want, not just the client id and client secret
https://github.com/nextauthjs/next-auth/blob/main/src/providers/fusionauth.js
